src/query.rs

```diff
@@ -38,20 +35,14 @@ impl Query {
         let ch = self.query.remove(self.index-1);
         self.index -= 1;
         self.pos -= if ch.len_utf8() > 1 {2} else {1};
-        return true;
+        true
     }
 
-    pub fn backward_char(&mut self) -> bool {
-        if self.index <= 0 { return false; }
-
-        match self.query.get(self.index-1) {
-            Some(ch) => {
-                self.index -= 1;
-                self.pos -= if ch.len_utf8() > 1 {2} else {1};
-            }
-            None => {}
+    pub fn backward_char(&mut self) {
+        if let Some(ch) = self.query.get(self.index-1) {
```

Comment:
I believe when `self.index` is `0`, `self.index-1` will cause `subtract overflow`.